tests/CollectionTest.php

```diff
@@ -259,6 +259,24 @@ public function testInsert(){
     $this->c->insert(-1, new TestClassA(5));
   }
 
+  public function testInsertEnd(){
+    $this->c->add(new TestClassA(1));
+    $this->c->add(new TestClassA(2));
```

Comment:
These aren't modifying the current collection as expected. These tests are incorrect.